src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -17,171 +20,193 @@ public class HttpConnectionDispatcher
+                RegisterLongPollingDisconnect(context, longPolling);
 
-                    // Raise OnConnected for new connections only since polls happen all the time
-                    if (isNewConnection)
-                    {
-                        state.Connection.Metadata["transport"] = "poll";
-                        state.Connection.Metadata.Format = format;
-                        state.Connection.User = context.User;
+                // Start the transport
+                var transportTask = longPolling.ProcessRequestAsync(context);
 
-                        // REVIEW: This is super gross, this all needs to be cleaned up...
-                        state.Close = async () =>
-                        {
-                            state.Connection.Channel.Dispose();
+                // Raise OnConnected for new connections only since polls happen all the time
+                var endpointTask = state.Connection.Metadata.Get<Task>("endpoint");
+                if (endpointTask == null)
+                {
+                    _logger.LogDebug("Establishing new Long Polling connection: {0}", state.Connection.ConnectionId);
 
-                            await endpointTask;
-                        };
+                    // This will re-initialize formatType metadata, but meh...
+                    InitializePersistentConnection(state, "poll", context, endpoint, format);
 
-                        endpointTask = endpoint.OnConnectedAsync(state.Connection);
-                        state.Connection.Metadata["endpoint"] = endpointTask;
-                    }
-                    else
+                    // REVIEW: This is super gross, this all needs to be cleaned up...
+                    state.Close = async () =>
                     {
-                        // Get the endpoint task from connection state
-                        endpointTask = state.Connection.Metadata.Get<Task>("endpoint");
-                    }
+                        state.Dispose();
 
-                    var resultTask = await Task.WhenAny(endpointTask, transportTask);
+                        await endpointTask;
+                    };
 
-                    if (resultTask == endpointTask)
-                    {
-                        // Notify the long polling transport to end
-                        state.Connection.Channel.Dispose();
+                    endpointTask = endpoint.OnConnectedAsync(state.Connection);
+                    state.Connection.Metadata["endpoint"] = endpointTask;
+                }
+                else
+                {
+                    _logger.LogDebug("Resuming existing Long Polling connection: {0}", state.Connection.ConnectionId);
+                }
 
-                        await transportTask;
-                    }
+                var resultTask = await Task.WhenAny(endpointTask, transportTask);
+
+                // Observe any exception
+                resultTask.GetAwaiter().GetResult();
```

Comment:
I think this is wrong. if the resultTask is a faulted endpointTask you will not release a pending poll. This is why we await endPointTask in the Close (which is actually broken too). 

I have a PR that sorts this out (https://github.com/aspnet/SignalR/pull/84)